### DISCUSSION TITLE
Reverse route warning

### DISCUSSION THREAD

**Discussion by airald** [devemux86/cruiser#306] [Q&A] [answered]
Cruiser offers off route warnings. It's very useful, but I think that there's another similar situation that could be added to this warning. If you are driving reverse, you are over the route but you are not following the route. This situation is maybe more important. When you see the map it seems that you are on the route, but depending on re-route settings, you don't hear any navigation instructions.

When POI enabled you can see that distance to next POI increases, you can imagine that something is wrong, but if not, could be complicated to see what happens.

It can happen easily when a route crosses itself. If the crossing is not evident, you can take the wrong way without Cruiser notice it to you.

**Comment by devemux86** [comment]
@airald have you enabled the "Settings | Navigation | Recalculate route | In case of reverse direction"?

After moving in the reverse direction for a while, the navigation recognizes it as an off-route situation with warnings etc.

**Reply by devemux86** [reply]
Or do you mean a separate warning when moving in the reverse direction?

**Reply by airald** [reply]
For me, can be in the same warning, both of them are out of route for me.

**Reply by 0709wiwiwi** [reply]
@airald 

> If the crossing is not evident, you can take the wrong way without Cruiser notice it to you.

A fine extra tool at complex crossing situations is to observe the correct track direction to go by:
Settings > Navigation > Advanced > Route appearance: "Show only the active route segment"

**Reply by devemux86** [reply]
> can be in the same warning, both of them are out of route for me.

@airald then isn't it already available?
Moving in reverse direction triggers the optical + voice warning of route missed.

**Reply by airald** [reply]
I think it does not work in reverse. When testing nearby POI, It happened 2 times that I missed to invert the route when going home/work, and I thought that something was wrong because curiser don't announces POI because distance increases due I was on reverse route. Route missed speech and also beep works when I don't follow it (I use it since was created)

Today I will test again.

**Reply by devemux86** [reply, marked as answer]
Please check again and also post the route so I can test it.

You need the have the rerouting reverse direction option enabled.
And check what you have in "Location | Advanced | Orientation threshold", e.g. 10 km/h.

Also it does not happen immediately, it takes some seconds to be sure.
Like in all off route cases, as GPS may report wrong location / heading.

**Reply by airald** [reply]
I never use reroute, just do it manually when needed.

**Reply by devemux86** [reply]
The aforementioned options are to detect the various off-route cases.

After the detection, whether to do a rerouting or not is always a user action, chosen by us.

**Reply by 0709wiwiwi** [reply]
Menu > in case of reverse direction = (un)set. I wonder if offering this item to be able to unset is necessary.

**Reply by airald** [reply]
I enabled reverse rerouting and keep disabled reroute. Now never reroute automatically and alerts me when going reverse. Exactly as I like. 

My error was think that cruiser will reroute on reverse just enabling it's setting (with general reroute disabled). Thanks for the explanation.

**Reply by devemux86** [reply]
@airald thanks for the feedback!

**Reply by 0709wiwiwi** [reply]
Menu > in case of reverse direction = (un)set.
perhaps it is better this setting is set active by default on the Cruiser production to come

**Reply by devemux86** [reply]
> perhaps it is better this setting is set active by default on the Cruiser production versions to come.

By default the reverse direction setting is enabled.

**Reply by bios4** [reply]
> I enabled reverse rerouting and keep disabled reroute. Now never reroute automatically and alerts me when going reverse. Exactly as I like.
> 
> My error was think that cruiser will reroute on reverse just enabling it's setting (with general reroute disabled). Thanks for the explanation.

> The aforementioned options are to detect the various off-route cases.
>
> After the detection, whether to do a rerouting or not is always a user action, chosen by us.

Indeed the wording "recalculate" is then not self-explaining for that functionality - both in the English and German translation.

Therefore I'd like to suggest the following changes:
"Recalculate route in case of reverse direction" --> "Alert in case of reverse direction" and
"Route neu berechnen falls in Gegenrichtung" --> "Warnen falls in Gegenrichtung"

What do you think @devemux86?

**Reply by devemux86** [reply]
Sure, we can change it.

@bios4 thanks again for continuing to suggest improvements to the app texts! :+1:

**Reply by WolfgangZ42** [reply]
If you specify "Recalculate route in navigation" and "Recalculate route in case of reverse direction" it's a **_Recalculate_** not an **_Alert_** ...

**Reply by bios4** [reply]
...but it isn't a **_Recalculate_**, therefore my request to change the text. 😉

**Reply by devemux86** [reply]
Recalculate route has its own on / off switch.
Below are options, I will put then in a titled border to better separate them.

Similarly in the auto zoom settings, we have a top switch and below options.
